packages/isar_core/src/native/isar_deserializer.rs

```diff
@@ -185,3 +185,877 @@ impl<'a> IsarDeserializer<'a> {
+            }
+
+            let values: Vec<&[u8]> = vec![&[], &[]];
+
+            let mut bytes = vec![0x6, 0x0, 0x0, 0x6, 0x0, 0x0, 0x9, 0x0, 0x0];
+            for &value in &values {
+                bytes.extend_from_slice(&[value.len() as u8, 0, 0]);
+                bytes.extend_from_slice(value);
+            }
+
+            let deserializer = IsarDeserializer::from_bytes(&bytes);
+            for (i, &expected) in values.iter().enumerate() {
+                let offset = (i * 3) as u32;
+                let deserialized_value = deserializer.read_dynamic(offset).unwrap();
+                assert_eq!(deserialized_value, expected);
+            }
+
+            let values: Vec<&[u8]> = vec![&[0x1; 10], &[0x2; 20], &[0x3; 30]];
+
+            let mut bytes = vec![0x9, 0x0, 0x0, 0x9, 0x0, 0x0, 0x16, 0x0, 0x0, 0x2d, 0x0, 0x0];
+            for &value in &values {
+                bytes.extend_from_slice(&[value.len() as u8, 0, 0]);
+                bytes.extend_from_slice(value);
+            }
+
+            let deserializer = IsarDeserializer::from_bytes(&bytes);
+            for (i, &expected) in values.iter().enumerate() {
+                let offset = (i * 3) as u32;
+                let deserialized_value = deserializer.read_dynamic(offset).unwrap();
+                assert_eq!(deserialized_value, expected);
+            }
+        }
+        #[test]
+        fn test_read_multiple_string() {
+            let values = ["foo", "bar"];
+
+            let mut bytes = vec![0x6, 0x0, 0x0, 0x6, 0x0, 0x0, 0xc, 0x0, 0x0];
+
+            for &value in &values {
+                bytes.extend_from_slice(&[value.len() as u8, 0, 0]);
+                bytes.extend_from_slice(value.as_bytes());
+            }
+
+            let deserializer = IsarDeserializer::from_bytes(&bytes);
+
+            for (i, &expected) in values.iter().enumerate() {
+                let offset = (i * 3) as u32;
+                let deserialized_value = deserializer.read_string(offset).unwrap();
+                assert_eq!(deserialized_value, expected);
+            }
+
+            let values = ["", ""];
+            let mut bytes = vec![0x6, 0x0, 0x0, 0x6, 0x0, 0x0, 0x9, 0x0, 0x0];
+            for &value in &values {
+                bytes.extend_from_slice(&[value.len() as u8, 0, 0]);
+                bytes.extend_from_slice(value.as_bytes());
+            }
+
+            let deserializer = IsarDeserializer::from_bytes(&bytes);
+
+            for (i, &expected) in values.iter().enumerate() {
+                let offset = (i * 3) as u32;
+                let deserialized_value = deserializer.read_string(offset).unwrap();
+                assert_eq!(deserialized_value, expected);
+            }
+        }
+    }
+
+    mod nested {
+        use super::*;
+
+        #[test]
+        fn test_read_nested() {
+            let null_int = NULL_INT;
+            let bytes: &[u8] = &[
+                0x3,
+                0x0,
+                0x0,
```

Comment:
I think the static size should be greater than 3

Reply:
https://github.com/isar/isar/blob/isar4/packages/isar_core/src/native/isar_serializer.rs#L1060